packages/@aws-cdk/cx-api/lib/placeholders.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Placeholders which can be used manifests
+ *
+ * These can occur both in the Asset Manifest as well as the general
+ * Cloud Assembly manifest.
+ */
+export class EnvironmentPlaceholders {
+  /**
+   * Insert this into the destination fields to be replaced with the current region
+   */
+  public static readonly CURRENT_REGION = '${AWS::Region}';
+
+  /**
+   * Insert this into the destination fields to be replaced with the current account
+   */
+  public static readonly CURRENT_ACCOUNT = '${AWS::AccountId}';
+
+  /**
+   * Insert this into the destination fields to be replaced with the current partition
+   */
+  public static readonly CURRENT_PARTITION = '${AWS::Partition}';
+
+  /**
+   * Replace the environment placeholders in all strings found in a complex object.
+   *
+   * Duplicated between cdk-assets and aws-cdk CLI because we don't have a good single place to put it
+   * (they're nominally independent tools).
+   */
+  public static replace(object: any, values: EnvironmentPlaceholderValues): any {
+    return this.recurse(object, value => {
+      value = replaceAll(value, EnvironmentPlaceholders.CURRENT_REGION, values.region);
+      value = replaceAll(value, EnvironmentPlaceholders.CURRENT_ACCOUNT, values.accountId);
+      value = replaceAll(value, EnvironmentPlaceholders.CURRENT_PARTITION, values.partition);
+      return value;
+    });
+  }
+
+  /**
+   * Like 'replace', but asynchronous
+   */
+  public static async replaceAsync(object: any, provider: IEnvironmentPlaceholderProvider): Promise<any> {
```

Comment:
Do we need this in cx-api? Is it used by the framework??

Reply:
It's used by both cdk-assets and aws-cli, and we have no other place to put it.